### PR TITLE
add more debug info for TypeInfoConverter

### PR DESCRIPTION
**Pull request by beyond-up**
https://github.com/bytedance/bitsail/issues/417

Signed-off-by:

## Pre-Checklist

Note: Please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/bytedance/bitsail/blob/master/docs/contributing.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests.

## Purpose

add more debug info and add check null  for TypeInfoConverter 

## Related Issues

(https://github.com/bytedance/bitsail/issues/417)

## New Behavior (screenshots if needed)

N/A

### CODE REVIEW

**Review by healchow — APPROVED** [review]

File: bitsail-common/src/main/java/com/bytedance/bitsail/common/type/filemapping/MongoTypeInfoConverter.java [hunk 1]
```diff
@@ -111,7 +121,12 @@ public TypeInfo<?> getTypeInfo(String type) {
   }
 
   public TypeInfo<?> getBasicTypeInfoFromMongoDBType(String mongoType) {
-    return reader.getToTypeInformation().get(mongoType);
```
Comment (lichang-bd) [inline bitsail-common/src/main/java/com/bytedance/bitsail/common/type/filemapping/MongoTypeInfoConverter.java:114]: suggest to add some ut test

Comment (beyond-up) [reply bitsail-common/src/main/java/com/bytedance/bitsail/common/type/filemapping/MongoTypeInfoConverter.java:114]: > @beyond-up Can we add more check after the code both in the `NativeFlinkTypeInfoUtil` and `ColumnFlinkTypeInfoUtil`
> 
> ```
>   public static RowTypeInfo getRowTypeInformation(List<ColumnInfo> columnInfos,
>                                                   TypeInfoConverter typeInfoConverter) {
> 
>     String[] fieldNames = new String[columnInfos.size()];
>     TypeInformation<?>[] fieldTypes = new TypeInformation[columnInfos.size()];
> 
>     for (int index = 0; index < columnInfos.size(); index++) {
>       String type = StringUtils.lowerCase(columnInfos.get(index).getType());
>       String name = columnInfos.get(index).getName();
> 
>       TypeInfo<?> typeInfo = typeInfoConverter.fromTypeString(type);
> 
>       //in here
>       fieldNames[index] = name;
>       fieldTypes[index] = toNativeFlinkTypeInformation(typeInfo);
>     }
> 
>     return new RowTypeInfo(fieldTypes, fieldNames);
>   }
> ```
> 
> Not every new type converter has enough validation rules, so add validation check after the `typeInfoConverter` invoke maybe more make sense? What do you think.

OK, no problem, then I will remove the exception information in the existing converter  like this

Comment (beyond-up) [reply bitsail-common/src/main/java/com/bytedance/bitsail/common/type/filemapping/MongoTypeInfoConverter.java:114]: > suggest to add some ut test

ok, i will add @lichang-bd

**Comment by beyond-up** [comment]
> 

ok, thanks

File: bitsail-cores/bitsail-core-flink-base/src/main/java/com/bytedance/bitsail/flink/core/typeutils/ColumnFlinkTypeInfoUtil.java [hunk 2]
```diff
@@ -35,17 +40,23 @@
...
+    LOG.debug("TypeInfoConverter = {}.", converter);
     String[] fieldNames = new String[columnInfos.size()];
     TypeInformation<?>[] fieldTypes = new TypeInformation[columnInfos.size()];
     for (int index = 0; index < columnInfos.size(); index++) {
       String type = StringUtils.lowerCase(columnInfos.get(index).getType());
       String name = columnInfos.get(index).getName();
 
       TypeInfo<?> typeInfo = converter.fromTypeString(type);
+      if (Objects.isNull(typeInfo)) {
+        throw BitSailException.asBitSailException(CommonErrorCode.UNSUPPORTED_COLUMN_TYPE,
```
[the remark this answers is not in the document]
Comment (beyond-up) [reply bitsail-cores/bitsail-core-flink-base/src/main/java/com/bytedance/bitsail/flink/core/typeutils/ColumnFlinkTypeInfoUtil.java:57]: > Plz add field name to the exception to let user quick to know.

ok, ths for your suggestion

File: bitsail-cores/bitsail-core-flink-base/src/main/java/com/bytedance/bitsail/flink/core/typeutils/NativeFlinkTypeInfoUtil.java [hunk 3]
```diff
@@ -49,6 +53,10 @@ public static RowTypeInfo getRowTypeInformation(List<ColumnInfo> columnInfos,
       String name = columnInfos.get(index).getName();
 
       TypeInfo<?> typeInfo = typeInfoConverter.fromTypeString(type);
+      if (Objects.isNull(typeInfo)) {
```
[the remark this answers is not in the document]
Comment (beyond-up) [reply bitsail-cores/bitsail-core-flink-base/src/main/java/com/bytedance/bitsail/flink/core/typeutils/NativeFlinkTypeInfoUtil.java:56]: I modified please look again, thx! @hk-lrzy

**Comment by beyond-up** [comment]
> @beyond-up Good job, i will merge it after CICD.

@hk-lrzy  Thank you for your guidance and patience !

**Pull request merged**